packages/loopback-connector-composer/lib/businessnetworkconnector.js

```
@@ -310,12 +310,25 @@ class BusinessNetworkConnector extends Connector {
                                 return [ result ];
                             });
 
-                    } else {
+                    } else if(objectId){
                         return registry.get(objectId)
                             .then((result) => {
                                 debug('Got Result:', result);
                                 return [ this.serializer.toJSON(result) ];
                             });
+                    } else {
+                        const queryString = FilterParser.parseFilter(filter, composerModelName);
+                        const query = networkConnection.buildQuery(queryString);
+                        if(typeof query === 'undefined'|| query === null) {
```

This shouldn't be possible; `buildQuery` should throw an error if `queryString` is invalid. You should not need this `if`.

---

@sstone1 @nklincoln  Thank you both for reviewing the code and I have updated the code per your comments.  Please review them again.